### ISSUE TITLE
Cosmos DB: Allow custom ID property

### ISSUE THREAD

**Issue by DaleyKD** [dotnet/efcore#21267]
My EF model is currently described as the following:
```csharp
public abstract class BaseEntity : IBaseEntity, IAuditEntity
{
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [JsonProperty("id")]
        public Guid ID { get; set; }

        [Required]
        [JsonProperty("partKey")]
        public string PartitionKey { get; set; } = string.Empty;

        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [JsonProperty("c")]
        public DateTime Created { get; set; }

        [Required]
        [JsonProperty("m")]
        public DateTime Modified { get; set; }

        public string Discriminator { get; set; }
}
```

I have to manually do `entity.Property<string>("id").CurrentValue = itemEntry.ID.ToString();` so that the EF Core shadow property "id" doesn't use `IdValueGenerator`.

I don't want my ID property to be a string.  I don't mind having a backing field be a string if we can find a way to prevent me manually changing the <string>("id") value every time I create an item.

**Comment by ajcvickers** [comment]
Following the merge of https://github.com/dotnet/efcore/issues/17751, this should now be possible with a value converter. This change should be in the daily builds shortly, and will be released in 5.0 preview 7.

**Comment by DaleyKD** [comment]
Thanks!  It may be a while before we go to EF Core 5 (even after it's official release), but it's good to know that it's coming.
